src/query-key-builder.test.ts

```diff
@@ -0,0 +1,339 @@
+import { Service } from 'basketry';
+import { QueryKeyBuilderFile } from './query-key-builder';
+import { NamespacedReactQueryOptions } from './types';
+
+describe('QueryKeyBuilderFile', () => {
+  const createService = (interfaces: any[]): Service => ({
+    kind: 'Service',
+    basketry: '1.1-rc',
+    title: { value: 'TestService' },
+    majorVersion: { value: 1 },
+    sourcePath: '',
+    interfaces,
+    types: [],
+    enums: [],
+    unions: [],
+  });
+
+  const createInterface = (name: string, methods: any[]) => ({
+    kind: 'Interface' as const,
+    name: { value: name },
+    methods,
+    protocols: {
+      http: [],
+    },
+  });
+
+  const createMethod = (
+    name: string,
+    parameters: any[] = [],
+    httpMethod: string = 'GET',
+  ) => ({
+    kind: 'Method' as const,
+    name: { value: name },
+    parameters,
+    returnType: undefined,
+    security: [],
+  });
+
+  const createParameter = (name: string, required = true) => ({
+    kind: 'Parameter' as const,
+    name: { value: name },
+    typeName: { value: 'string' },
+    isArray: false,
+    isPrimitive: true,
+    description: null,
+    deprecated: null,
+    errors: [],
+    warnings: [],
+    sourcePath: '',
+    rules: [
+      ...(required
+        ? [
+            {
+              kind: 'Rule' as const,
+              id: 'required',
+              value: null,
+              errors: [],
+              warnings: [],
+              sourcePath: '',
+            },
+          ]
+        : []),
+    ],
+  });
+
+  describe('QueryKeyMap generation', () => {
+    it('generates correct interface structure', () => {
+      const service = createService([
+        createInterface('Widget', [
+          createMethod('getWidgets', [createParameter('status', false)]),
+          createMethod('getWidgetById', [createParameter('id')]),
+        ]),
+        createInterface('Gizmo', [createMethod('getGizmos')]),
+      ]);
+
+      const builder = new QueryKeyBuilderFile(service, {});
+      const output = Array.from(builder.build()).join('\n');
+
+      expect(output).toContain('export interface QueryKeyMap {');
+      expect(output).toContain('widget: {');
+      expect(output).toContain('getWidgets: GetWidgetsParams | undefined;');
+      expect(output).toContain('getWidgetById: GetWidgetByIdParams;');
+      expect(output).toContain('gizmo: {');
+      expect(output).toContain('getGizmos: GetGizmosParams | undefined;');
```

Comment:
This is fine. I want to find a solution like this to unit testing generator outputs other than just snapshot tests. The only issue I see with this is testing indentation (not critical) and line order.